File: android/src/main/java/com/lightningkite/khrysaliswebsite/layouts/HomeXml.kt
//
// HomeXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.khrysaliswebsite.layouts

import android.widget.*
import android.view.*
import com.lightningkite.khrysalis.views.*
import com.lightningkite.khrysaliswebsite.R

class HomeXml {

    lateinit var featureList: LinearLayout
    lateinit var tutorial: Button
    lateinit var docs: Button
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ViewDependency): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.home, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        featureList = view.findViewById<LinearLayout>(R.id.feature_list)
        tutorial = view.findViewById<Button>(R.id.tutorial)
        docs = view.findViewById<Button>(R.id.docs)
        
        
        
        return view
    }
}

File: web/src/layout/HomeXml.ts
//
// HomeXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId } from 'khrysalis/dist/views/html'
import { customViewSetDelegate } from 'khrysalis/dist/views/CustomView.actual'
import htmlForDefault from './home.html'
//! Declares com.lightningkite.khrysaliswebsite.layouts.HomeXml
export class HomeXml {
    xmlRoot!: HTMLElement;
    featureList!: HTMLDivElement;
    tutorial!: HTMLButtonElement;
    docs!: HTMLButtonElement;
    loadHtmlString(): string {
        return htmlForDefault;
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString());
        this.xmlRoot = view
        this.featureList = getViewById<HTMLDivElement>(view, "feature_list");
        this.tutorial = getViewById<HTMLButtonElement>(view, "tutorial");
        this.docs = getViewById<HTMLButtonElement>(view, "docs");
        return view
    }
}
